### DISCUSSION TITLE
Struggling with project build logic....

### DISCUSSION THREAD

**Discussion by toddabraham** [smartcontractkit/full-blockchain-solidity-course-py#249] [Q&A]
Hi @PatrickAlphaC ,

I'm not sure if my discussion label is accurate, but let's see.

I'm on Lesson 13, and found myself facing the same error when trying to run deploy.py as I faced in Lesson 11 (https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/129#discussion-3604050).

To me this highlights a hole in my understanding of what components need to be in place when building a project.

What areas should I be going back and reviewing to help fill this cognitive void in my learning?

![image](https://user-images.githubusercontent.com/26919212/138542936-a5e3d2e3-d07a-4c65-b550-c523177ff3d5.png)

All the folders listed above were copied from Lesson 13 Github  repo, which concerns me as I feel I should have a better understanding of why we setup these mock contracts in a test folder inside contracts.

I'm rambling now, but I hope you understand my meaning here.  Just trying to solidify my learning.

**Comment by cromewar** [comment]
Remember every time you add a new contract you should compile the whole project again, this allow brownie to add the contract into the `build` folder and access to it, could you please paste here a screenshot of your build folder?

**Reply by toddabraham** [reply]
Thanks @cromewar.

Here's my build folder:

![image](https://user-images.githubusercontent.com/26919212/138561087-f936e6dc-0832-4986-bdb2-353d75f79b23.png)

**Reply by cromewar** [reply]
Actually i want to see this one `build`
![image](https://user-images.githubusercontent.com/20199180/138567093-9e8b3e54-f662-4e32-9b80-6afa44e895b9.png)
did you try deleting it and compile again?

**Reply by toddabraham** [reply]
Here you go.   
![image](https://user-images.githubusercontent.com/26919212/138576020-85c3477e-3fe5-4bf8-8710-aa2aaf50fe3c.png)

But at the moment I have no problems after creating all the test.sol files...

![image](https://user-images.githubusercontent.com/26919212/138576069-73e8f994-b2fa-4b68-9009-31c2e47b6c20.png)

**Reply by cromewar** [reply]
Did the deploy script work fine? 

Regarding to the highlights: this is because of the VSCode extension, for example let's suppose your compiler version is 0.8.0, and you import a test contract which uses 0.6.6 version, in most of the cases the version will come with this  `^` symbol like this:

`pragma solidity ^0.6.6;`

Meaning that you can use any further version of solidity, and the compiler knows it. However the Vscode extension doesn't, if you use any other version the extension is going to complain about it... you can ignore this for now though. Hopefully the extension will get better with time.
